### PR TITLE
[consensus V2] minor updates to packer

### PR DESCRIPTION
**Pull request by tarakby**
This PR should be merged after https://github.com/onflow/flow-go/pull/1604 (based on the same branch). 

It adds minor updates to the packer implementation:
- does not rely on the constants `SigTypeStaking` and `SigTypeRandomBeacon` being 0 and 1. The implementation should be agnostic of the constant value.
- minor optimizations in the bit operations. 

@zhangchiqing  The code of the bit serialization/deserialization can be simpler if we choose the sigType at index 0 to be encoded in the bit at index 0 (little endian). Currently, the sigType at index 0 is encoded at index 7. Is there a reason for this encoding choice?

### CODE REVIEW

File: consensus/hotstuff/signature/packer.go [hunk 1]
```diff
@@ -158,88 +158,86 @@ func (p *ConsensusSigDataPacker) Unpack(blockID flow.Identifier, signerIDs []flo
...
-		b := byte(0)
-		offset := 7
-		for pos := byt; pos < byt+8 && pos < len(sigTypes); pos++ {
-			sigType := sigTypes[pos]
-
-			if !sigType.Valid() {
-				return nil, fmt.Errorf("invalid sig type: %v at pos %v", sigType, pos)
-			}
-
-			b ^= byte(sigType) << offset
```
Comment (tarakby) [inline consensus/hotstuff/signature/packer.go:194]: This line rely on `sigType` constants being `1` or `0` while there is no guarantee for it.

File: consensus/hotstuff/signature/packer.go [hunk 2]
```diff
@@ -158,88 +158,86 @@ func (p *ConsensusSigDataPacker) Unpack(blockID flow.Identifier, signerIDs []flo
...
 	// parse each bit in the bit-vector, bit 0 is SigTypeStaking, bit 1 is SigTypeRandomBeacon
+	var byt byte
+	var offset int
 	for i := 0; i < count; i++ {
-		byt := serialized[i/8]
-		offset := 7 - (i % 8)
-		posMask := byte(1 << offset)
-		if byt&posMask == 0 {
+		byt = serialized[i>>3]
+		offset = 7 - (i & 7)
```
Comment (zhangchiqing) [inline consensus/hotstuff/signature/packer.go:227]: 👍

**Review by zhangchiqing — APPROVED** [review]

**Review by AlexHentschel — APPROVED** [review]
very nice. Like the consistent utilization of bit shifts, addition, and subtraction operations; much more efficient than modulo.

**Comment by tarakby** [comment]
bors merge

**Pull request merged**